### PR TITLE
Proofread specification based testing

### PR DESCRIPTION
**Pull request by HellboundHound**
Changes include:

- Rephrasing of Chocolate Bars example and Exercise 7
- Corrected the spelling of "divisable" to "divisible"
- Restructured several sentences

Kind regards,
Danny Breunissen

### CODE REVIEW

File: chapters/testing-techniques/specification-based-testing.md [hunk 1]
```diff
@@ -166,9 +166,9 @@ After the method returns a value, we assert that this is the expected value.
 
 So far we have derived partitions by just looking at the specification of the program.
 We basically used our experience and knowledge to derive the test cases.
-We now go over a more systematic way of deriving these partitions: the **Category-Partition** method.
+We will discuss a more systematic way of deriving these partitions: the **Category-Partition** method.
```
Comment (mauricioaniche) [inline chapters/testing-techniques/specification-based-testing.md:169]: Maybe, "In this chapter, we will discuss"?

File: chapters/testing-techniques/specification-based-testing.md [hunk 2]
```diff
@@ -219,14 +219,16 @@ Each of the test cases corresponds to one of the partitions that we want to test
...
-> There are small bars (1 kilo each) and big bars (5 kilos each). 
-> We should calculate the number of small bars to use, 
-> assuming we always use big bars before small bars. Return -1 if it is impossible.
 >
-> The input of the program is thus the number of small bars, the number of big bars,
-> and the total number of kilos to store.
+> A package can store a certain amount of chocolate bars in kilos.
+> There are small bars (1 kilo each) and big bars (5 kilos each).
+> Assume that the package is always filled with big bars first, 
+> if there's room left, the package will then be topped of with small bars.
```
Comment (mauricioaniche) [inline chapters/testing-techniques/specification-based-testing.md:226]: I agree that the requirement is too short. But I think the new phrasing is a bit ambiguous. 

The "if there's room left" brings some ambiguity, as it's not "if", the box _has_ to be filled up to the _total_ number of kilos. 

What about:

A package stores a certain number of chocolate bars in kilos.
A package is composed of small bars (1 kilo each) and big bars (5 kilos each).

Assume that the package is always filled with the maximum number of big bars possible, return the number of small bars required to complete the package. Return -1 if it is not possible to fill the package completely.

The input of the program is: the number of available small bars, the number of available big bars, and the total number of kilos of the package.

What do you think?

File: chapters/testing-techniques/specification-based-testing.md [hunk 3]
```diff
@@ -340,9 +342,9 @@ What is an Equivalence Partition?
 We have a program called FizzBuzz.
 It does the following:
 > Given an integer `n`, return the string form of the number followed by `"!"`.
-> If the number is divisable by 3 use `"Fizz"` instead of the number,
-> and if the number is divisable by 5 use `"Buzz"` instead of the number,
-> and if the number is divisable by both 3 and 5, use `"FizzBuzz"`
+> If the number is divisible by 3 use `"Fizz"` instead of the number,
```
Comment (mauricioaniche) [inline chapters/testing-techniques/specification-based-testing.md:345]: Great catch!

File: chapters/testing-techniques/specification-based-testing.md (hunk 2 above)
Comment (HellboundHound) [reply chapters/testing-techniques/specification-based-testing.md:226]: I was struggeling with the phrasing as well to be honest. I think your amends make it less ambiguous. Will update the PR.

File: chapters/testing-techniques/specification-based-testing.md (hunk 1 above)
Comment (HellboundHound) [reply chapters/testing-techniques/specification-based-testing.md:169]: Added your suggestion!

**Pull request merged**